### ISSUE TITLE
Can't start the session

### ISSUE THREAD

**Issue by bglowacki** [neo4jrb/neo4j-core#311]
Getting this error when I try to do the example from Readme
```
/Users/bartek/.rvm/rubies/ruby-2.4.2/bin/ruby -e $stdout.sync=true;$stderr.sync=true;load($0=ARGV.shift) /Users/bartek/Projects/tmp/lib/neo4j_exporter/export_tags.rb
/Users/bartek/.rvm/gems/ruby-2.4.2/gems/neo4j-core-8.1.0/lib/neo4j/core/cypher_session/adaptors/bolt.rb:144:in `open_socket': uninitialized constant Neo4j::Core::CypherSession::Adaptors::Bolt::TCPSocket (NameError)
	from /Users/bartek/.rvm/gems/ruby-2.4.2/gems/neo4j-core-8.1.0/lib/neo4j/core/cypher_session/adaptors/bolt.rb:27:in `initialize'
	from /Users/bartek/Projects/tmp/lib/neo4j_exporter/export_tags.rb:3:in `new'
	from /Users/bartek/Projects/tmp/lib/neo4j_exporter/export_tags.rb:3:in `<top (required)>'
	from -e:1:in `load'
	from -e:1:in `<main>'

Process finished with exit code 1
```

Additional information which could be helpful if relevent to your issue:

### Code example (inline, gist, or repo)
```
require "neo4j/core/cypher_session/adaptors/bolt"

bolt_adaptor = Neo4j::Core::CypherSession::Adaptors::Bolt.new('bolt://neo4j:neo4j@localhost:7687', timeout: 10)
neo4j_session = Neo4j::Core::CypherSession.new(bolt_adaptor)

neo4j_session.query('MATCH (n) RETURN n LIMIT 10')
```

### Runtime information:

Ruby version: ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin16]
Neo4j database version: 3.2
`neo4j-core` gem version:  8.1

**Comment by cheerfulstoic** [comment]
Are you running in a bare Ruby environment?  It could be that `require 'socket'` is needed (which would probably be included by default in something like a Rails environment, which is probably why this got missed).

If requiring `socket` fixes the problem for you I can add that to [bolt.rb](https://github.com/neo4jrb/neo4j-core/blob/master/lib/neo4j/core/cypher_session/adaptors/bolt.rb) which should fix the issue without you needed to do that

**Comment by bglowacki** [comment]
Requiring `socket` helped however the example did not work untill I required `neo4j` as well.

**Comment by cheerfulstoic** [comment]
Ok, released as 8.1.1.  I suspect needing to require `neo4j` is a different problem (though it should only be needed if you need `ActiveNode` / `ActiveRel`)
